sous_chef/sous_chef/recipe_book/read_recipe_book.py

```diff
@@ -31,11 +39,22 @@ class RecipeBook(DataframeSearchable):
     def __post_init__(self):
         # load basic recipe book to self.dataframe
         self.dataframe = self._read_recipe_book()
-
         if self.config.deduplicate:
             self._select_highest_rated_when_duplicated_name()
 
-    def get_recipe_by_title(self, title):
+    def get_random_recipe_by_category(self, category: str) -> Recipe:
+        return self._select_random_recipe_weighted_by_rating(
+            method_match=self._is_value_in_list,
+            field="categories",
+            search_term=category,
+        )
+
+    def get_random_recipe_by_tag(self, tag: str) -> Recipe:
```

Comment:
We could add coverage to these tests without assuming too much. These require that all recipe data has been loaded. i.e. when we search by title with postgres, i'm not sure that we'd want to load all recipes & not instead do a query for similar names...